docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+### The Worker
+
+As mentioned earlier, a special process called a "Worker" is responsible for actually executing pieces of Workflow code. This is what our Worker looks like:
+
+<!--SNIPSTART java-project-template-worker-->
+<!--SNIPEND-->
+
+Notice that the Worker will listen to the same Task Queue that the Workflow and Activity tasks are sent to. This is called "Task routing", and is a built-in mechanism for load balancing.
+
+<!--SNIPSTART java-project-template-shared-constants-->
+<!--SNIPEND-->
+
+Let's start the Worker. From within IntelliJ run the TransferMoneyWorkfer class, or from the project root, run the following command:
+
+```
+./gradlew startWorker
+```
+
+When the Worker starts it begins polling the Task Queue. The first Task it finds is the Workflow Task, which it executes. Executing the Workflow causes the Activity tasks to be sent to the Task Queue as well. The Worker then pulls each of the Activity Tasks in their respective order from the Task Queue and executes the corresponding Activities.
+
+Congratulations, you have run a Temporal Workflow application!
+
+## Failure simulation
+
+![Warning](/img/docs/warning.png)
+
+By running the application as is, at minimum you got a taste of one of Temporal's amazing value propositions: visibility into the Workflow and the status of the Workers executing the code. Let's explore another key value proposition, maintaining the state of a Workflow, by simulating some failures. Make sure your Worker process is stopped before proceeding.
+
+### Server crash
+
+In most modern application environments, to accommodate a host failure, leader election processes are applied to multiple application instances which are backed by databases that maintain the application state and so on. With Temporal, even if the Temporal server is down, the state of the Workflow is preserved. You can easily test this by following these steps:
+
+1. Start the Workflow again.
+2. Verify the Workflow is running in the UI.
+3. Shut down the Temporal server by either using 'Ctrl c' or via the Docker dashboard.
+4. After it has stopped, restart the Temporal server and visit the UI.
+
+The Workflow is still there!
+
+### Activity error
+
+Next let's simulate a bug in the Activity method. Inside your project find the AccountActivityImpl.java file and uncomment the line that throws an exception in the deposit method:
+
+<!--SNIPSTART java-project-template-activity-implementation-->
+<!--SNIPEND-->
+
+Save it and run the Workflow and the Worker. The Worker executed the Workflow, and completes the `withdraw()` Activity. But then the exception is thrown when it attempts the `deposit()` Activity. Notice how the Worker keeps retrying the Activity? Visit the UI and click on the RunId of the Workflow. You will see the pending Activity listed there with details such as its state, the number of times it has been attempted, and the next scheduled attempt.
```

Comment:
Is it possible to provide links to Temporal web from this tutorial? That way they can click the link and be in the right page of the UI?

Reply:
Unfortunately, the best we can do is link to localhost:8088 - the run id changes everytime.